### PR TITLE
Add some openshift_node_group and openshift_node_group_name docs

### PR DESCRIPTION
**Pull request by sdodson**
/cc @michaelgugino

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,6 +94,43 @@ cd openshift-ansible
...
+      - 'node-role.kubernetes.io/infra=true'
+    edits: []
+  - name: node-config-compute
+    labels:
+      - 'node-role.kubernetes.io/compute=true'
+    edits: []
+```
+
+When configuring this in the INI based inventory this must be translated into
+JSON. Here's an example of a group named `node-config-all-in-one` which is
```
Comment (michaelgugino) [inline README.md]: This is not actually json syntax and is not compatible with json.  It's python style dictionary / list syntax.

File: README.md [hunk 2]
```diff
@@ -94,6 +94,43 @@ cd openshift-ansible
 sudo ansible-playbook -i inventory/hosts.localhost playbooks/prerequisites.yml
 sudo ansible-playbook -i inventory/hosts.localhost playbooks/deploy_cluster.yml
 ```
+## Node Group Definition and Mapping
+In 3.10 and newer all members of the [nodes] inventory group must be assigned a
```
Comment (michaelgugino) [inline README.md]: Need to add some bits about how node labels work, and what the configmap bits are, generally.

File: README.md (hunk 1 above)
Comment (sdodson) [reply README.md]: Ok, how's it look now?

File: README.md [hunk 3]
```diff
@@ -94,6 +94,45 @@ cd openshift-ansible
 sudo ansible-playbook -i inventory/hosts.localhost playbooks/prerequisites.yml
 sudo ansible-playbook -i inventory/hosts.localhost playbooks/deploy_cluster.yml
 ```
+## Node Group Definition and Mapping
+In 3.10 and newer all members of the [nodes] inventory group must be assigned an
+`openshift_node_group_name`. This value is used to select the configmap that
+configures each node. By default there are three node groups defined
```
Comment (michaelgugino) [inline README.md]: By default, there are three configmaps created; one for each node group defined in openshift_node_groups: ...

File: README.md [hunk 4]
```diff
@@ -94,6 +94,49 @@ cd openshift-ansible
...
+kubeletArguments.pods-per-core set to 20
+
+```
+openshift_node_groups=[{'name': 'node-config-all-in-one', 'labels': ['node-role.kubernetes.io/master=true', 'node-role.kubernetes.io/infra=true', 'node-role.kubernetes.io/compute=true'], 'edits': [{ 'key': 'kubeletArguments.pods-per-core','value': ['20']}]}]
+```
+
+For upgrades, the upgrade process will block until you have the required
+configmaps in the openshift-node namespace. Please define
+`openshift_node_groups` as explained above or accept the defaults and run the
+playbooks/openshift-master/openshift_node_group.yml playbook to have them
```
Comment (ganhuang) [inline README.md]: @sdodson @michaelgugino Could we have an option in inventory to include the playbook specifically inside the upgrade entry playbook? e.g `openshift_node_groups_migration` defaults to `False`:

- When `openshift_node_groups_migration=False`, we abort to give the warnings as of now until corresponding configmap created. Two ways for advice for next step: 
  - 1. Create manually via document and run the upgrade playbook again.
  - 2. By enabling the option and add relevant parameters to run the upgrade playbook again.
- When `openshift_node_groups_migration=True`, we're going to create the configmap according to the settings in inventory (Using the default configurations if not set)

We don't want to add the complications for upgrade by introducing additional playbook. Some users might just need to use the default configurations to complete the upgrade.

We could also benefit by using one entry file for upgrade. I'm sorry so late to catch the changes.

**Comment by sdodson** [comment]
We'll discuss this during our standup today. Previously we've said that we'd require that configmaps are created before upgrade and we'd provide documentation and/or a tool to create them so that they may be manually created. Thinking a bit more about this, is there even an openshift-node namespace before we do the 3.10 upgrade? If we were to simply generate a configmap yaml file and let the admin load that I'd be concerned that they may not create the openshift-node namespace properly so we'd have to ensure that the yaml file included everything required to create the namespace.

**Comment by vrutkovs** [comment]
>Thinking a bit more about this, is there even an openshift-node namespace before we do the 3.10 upgrade?

Yes, `openshift-node` exists in 3.9. It only has `system:node-config-reader` role.

**Comment by sdodson** [comment]
Decision is that we'll stick with the code as is. This means that the upgrade will block until the requisite configmaps are created and it will link to documentation that more thoroughly explains node configuration specifics in 3.10 and outlines how to run the playbook to create the configmaps then advises the admin to carefuly review the configmaps then continue with the upgrade.  Due to the risk involved with the substantial change to configuration in 3.10 we want the admin to be fully aware of the details and have the opportunity to review before completing a migration which would potentially result in workloads being unschedulable.

**Pull request merged**